### ISSUE TITLE
Collation and charset not match error when migrating utf8mb4 databases

### ISSUE THREAD

**Issue by tiaozi0912** [sequelize/umzug#46]
In the config file define something like 
```
{
  "production": {
    ...,
   {
     "define": {
       "charset": "utf8mb4",
       "collate": "utf8mb4_general_ci",
     }
   }
  }
}
```
to work with utf8mb4.

When running migration, it throws the error
```
{ [SequelizeDatabaseError: ER_COLLATION_CHARSET_MISMATCH: COLLATION 'utf8_unicode_ci' is not valid for CHARACTER SET 'utf8mb4']
  name: 'SequelizeDatabaseError',
  message: 'ER_COLLATION_CHARSET_MISMATCH: COLLATION \'utf8_unicode_ci\' is not valid for CHARACTER SET \'utf8mb4\'',
  parent:
   { [Error: ER_COLLATION_CHARSET_MISMATCH: COLLATION 'utf8_unicode_ci' is not valid for CHARACTER SET 'utf8mb4']
     code: 'ER_COLLATION_CHARSET_MISMATCH',
     errno: 1253,
     sqlState: '42000',
     index: 0,
     sql: 'CREATE TABLE IF NOT EXISTS `SequelizeMeta` (`name` VARCHAR(255) NOT NULL UNIQUE , PRIMARY KEY (`name`)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE utf8_unicode_ci;' },
  original:
   { [Error: ER_COLLATION_CHARSET_MISMATCH: COLLATION 'utf8_unicode_ci' is not valid for CHARACTER SET 'utf8mb4']
     code: 'ER_COLLATION_CHARSET_MISMATCH',
     errno: 1253,
     sqlState: '42000',
     index: 0,
     sql: 'CREATE TABLE IF NOT EXISTS `SequelizeMeta` (`name` VARCHAR(255) NOT NULL UNIQUE , PRIMARY KEY (`name`)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE utf8_unicode_ci;' },
  sql: 'CREATE TABLE IF NOT EXISTS `SequelizeMeta` (`name` VARCHAR(255) NOT NULL UNIQUE , PRIMARY KEY (`name`)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE utf8_unicode_ci;' }
```

It is due to the collate is hardcoded as `utf8_unicode_ci` (https://github.com/sequelize/umzug/commit/37600173b872a9622cd61bb7335abb3937d66360) while it uses the charset that defined in define.charset in the config file.

Propose to explicitly set the charset as 'utf8' in https://github.com/sequelize/umzug/commit/37600173b872a9622cd61bb7335abb3937d66360 too. 

Would be happy to create a PR if this looks good.

**Comment by sdepold** [comment]
@fnimick ?

**Comment by fnimick** [comment]
This is odd, we have not had an issue in production migrating utf8mb4 databases - if you have a PR that resolves this, please submit it!

**Comment by tiaozi0912** [comment]
PR https://github.com/sequelize/umzug/pull/48

**Comment by jukkah** [comment]
@tiaozi0912 Fully fixed in #48?
